src/main/java/org/folio/rest/validator/ResourcePutBodyValidator.java

```diff
@@ -0,0 +1,91 @@
+    }
+    
+    ResourceDataAttributes attributes = request.getData().getAttributes();
+    boolean isSelected = attributes.getIsSelected().booleanValue();
+    String cvgStmt = attributes.getCoverageStatement();
+
+    /**
+     * Updates cannot be made to a resource unless it is selected
+     */
+    if(isSelected) {
+      /**
+       * Following fields can be updated only for a custom resource although UI sends complete payload
+       * for both managed and custom resources
+       */
+      if(isTitleCustom) {
+        String name = attributes.getName();
+        String pubType = attributes.getPublicationType() != null ? attributes.getPublicationType().value() : null;
+        String pubName = attributes.getPublisherName();
+        String edition = attributes.getEdition();
+        String description = attributes.getDescription();
+        String url = attributes.getUrl();
+        
+        ValidatorUtil.checkIsBlank("name", name);
+        ValidatorUtil.checkMaxLength("name", name, 400);
+        ValidatorUtil.checkIsBlank("publicationType", pubType);
+        if(!StringUtils.isBlank(pubName)) {
+          ValidatorUtil.checkMaxLength("publisherName", pubName, 250);
+        }
+        if(!StringUtils.isBlank(edition)) {
+          ValidatorUtil.checkMaxLength("edition", edition, 250);
+        }
+        if(!StringUtils.isBlank(description)) {
+          ValidatorUtil.checkMaxLength("description", description, 1500);
+        }
+        if(!StringUtils.isBlank(url)) {
+          ValidatorUtil.checkMaxLength("url", url, 600);
+          ValidatorUtil.checkUrlFormat("url", url);
+        }
+        if(Objects.nonNull(attributes.getIdentifiers())) {
+          attributes.getIdentifiers().forEach(identifier -> ValidatorUtil.checkIdentifierValid("identifier", identifier));
+        }
+      }
+
+      /**
+       * Following fields can be updated only for a managed resource although UI sends complete payload
+       * for both managed and custom resources
+       */
+      if(!StringUtils.isBlank(cvgStmt)) {
+        ValidatorUtil.checkMaxLength("coverageStatement", cvgStmt, 250);
+      }
+      if(Objects.nonNull(attributes.getCustomCoverages())) {
+        attributes.getCustomCoverages().forEach(customCoverage -> {
+          ValidatorUtil.checkDateValid("beginCoverage", customCoverage.getBeginCoverage());
+          ValidatorUtil.checkDateValid("endCoverage", customCoverage.getEndCoverage());
+        });  
+      }
+    } else {
+      Boolean isHidden = attributes.getVisibilityData() != null ? attributes.getVisibilityData().getIsHidden() : null;
+      EmbargoUnit embargoUnit = attributes.getCustomEmbargoPeriod() != null ? attributes.getCustomEmbargoPeriod().getEmbargoUnit() : null;
+      List<Coverage> customCoverages = attributes.getCustomCoverages();
+      if (!isTitleCustom && (!Objects.isNull(cvgStmt) || !Objects.isNull(embargoUnit) || isHidden || (!Objects.isNull(customCoverages) && !customCoverages.isEmpty()))) {
```

Comment:
if isHidden will be null then this will cause NPE, it should be checked for null, or default value should be changed from null to false